nyaa/api_torznab.py

```diff
@@ -101,71 +112,113 @@ def torznab_caps():
     return render_torznab_caps()
 
 
-def torznab_search():
-    page_size = app.config['RESULTS_PER_PAGE']
+def get_categories_filter(cat):
+    categories = []
+    if cat:
+        for item in cat.split(','):
+            # Parse 1_1 format for compatibility
```

Comment:
Personally I was opting for treating no categories the same as no desired categories (see Principle of least surprise).

If I have list, and filter out any undesired items, that leaves me with a list containing only desired items.  If an empty list is considered acceptable, why shouldn't a list with all items removed (thus an empty list) also be acceptable?